0004/README.md

```diff
@@ -0,0 +1,514 @@
+corruption of a build system, the exposure of a signing key, or a malicious
+change to a source repository. 
+
+### Expectations when publishing a package to a registry
+
+A developer submits a new version of a package to a registry. A policy evaluator
+at the registry verifies the signature on the SLSA provenance for that package,
+and verifies that the provenance meets the provenance expectations for this
+version of the package (one policy covers multiple versions of a specific
+package).
+
+Initially the policy evaluator will run pre-upload or post-upload, producing
+alerts only, with expectations stored at the evaluator. For enforcement the
+evaluator will need to be integrated into the package registry, and expectations
+may become managed centrally.
+
+A developer may also run a policy evaluator locally, with their own provenance
+expectations, before attempting to submit their package to a registry.
+
+An organization that uses a private registry may have different expectations and
+verification than a public registry.
+
+### Expectations when installing a package from a registry
+
+A developer installs a software package as an application or library dependency,
+or an end user installs an application package in a user directory. In both
+cases the npm package manager installs transitive dependencies for the specified
+package, and records each installed package in the local lock file
+(package-lock.json).
+
+A policy evaluator at the developer or end user applies a specific SLSA
+provenance policy for each package version that is recorded in the local lock
+file  (one policy covers multiple versions of a specific package).
+
+Initially the policy evaluator will run post-installation, producing alerts
+only, using expectations stored at the local evaluator instance. For enforcement
+the evaluator will need to be integrated into the package installer, and
+expectations may become managed centrally. 
+
+### Generating provenance expectations
+
+To overcome a bootstrapping problem, we assume that compromises will be rare,
+and propose to generate package-specific SLSA provenance expectations on first
+use. Each such ‘trust on first use' policy contains expectations based on
+observed provenance, and can be used as a baseline to detect unexpected changes,
+for example a too low SLSA level, missing provenance, or or an unexpected change
+in a ‘sensitive' provenance property.
+
+We expect that provenance expectations will initially be managed locally where
+they are used: at the package developer, at the package registry, or at the
+package user. We also expect that policies will initially be alerting only, i.e.
+not enforcing.
+
+As experience grows and confidence increases, we expect that policies will
+evolve to become ‘enforcing', and that expectations may become managed
+centrally.
+
+## Background: npm 
+
+[Npm](https://www.npmjs.com/) is the package manager for Node.js software
```

Comment:
```suggestion
[npm](https://www.npmjs.com/) is the package manager for Node.js software
```
AIUI npm is never capitalised.

Reply:
Yes, npm is always lower case: https://github.com/npm/cli#is-it-npm-or-npm-or-npm